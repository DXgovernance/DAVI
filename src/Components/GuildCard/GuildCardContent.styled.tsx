import styled from 'styled-components';

export const DaoIcon = styled.img`
  height: 4rem;
  width: 4rem;
`;

