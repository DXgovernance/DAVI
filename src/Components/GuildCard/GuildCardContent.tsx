import styled from 'styled-components';
import { Box } from 'Components/Primitives/Layout/index';

const GuildCardContent = styled(Box)`
  margin-top: 2rem;
  color: ${({ theme }) => theme.colors.text};
`;

export default GuildCardContent;
