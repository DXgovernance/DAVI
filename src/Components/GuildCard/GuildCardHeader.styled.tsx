import styled from 'styled-components';
import { Flex, Box } from 'Components/Primitives/Layout/index';

export const Header = styled(Flex)`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  color: ${({ theme }) => theme.colors.text};
`;

export const MemberWrapper = styled(Flex)`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  color: ${({ theme }) => theme.colors.card.grey};
`;

export const ProposalsInformation = styled(Box)<{ proposals?: string }>`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  margin: 0.5rem;
  border-radius: 15px;
  border: 1px solid
    ${({ proposals, theme }) =>
      proposals === 'active'
        ? theme.colors.card.green
        : theme.colors.card.grey};
  background-color: ${({ theme }) => theme.colors.background};
  color: ${({ proposals, theme }) =>
    proposals === 'active' ? theme.colors.card.green : theme.colors.card.grey};
  padding: 0.25rem 0.4rem;
`;
