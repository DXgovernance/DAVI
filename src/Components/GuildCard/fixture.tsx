import { GuildCardProps } from './types';

export const fullParameters: GuildCardProps = {
  guildAddress: '0xa47BbE8Dd6dB29D45FE5eeD838c4f136884AEAF3',
  numberOfMembers: 5,
  t: key => key,
  numberOfActiveProposals: 10,
  ensName: 'REPGuild',
  data: { name: 'REPGuild' },
};

export const noProposals: GuildCardProps = {
  guildAddress: '0xa47BbE8Dd6dB29D45FE5eeD838c4f136884AEAF3',
  numberOfMembers: 5,
  t: key => key,
  numberOfActiveProposals: 0,
  ensName: 'REPGuild',
  data: { name: 'REPGuild' },
};

export const loadingParameters: GuildCardProps = {
  isLoading: true,
  guildAddress: null,
  numberOfMembers: null,
  t: null,
  numberOfActiveProposals: null,
  ensName: null,
  data: null,
};
