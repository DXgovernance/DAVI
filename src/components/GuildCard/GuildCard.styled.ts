import { css } from 'styled-components';

export const cardWrapperStyles = css`
  margin-bottom: 1rem;
  box-shadow: 0px 4px 8px 0px rgba(0, 0, 0, 0.2);
  width: 18%;
  min-width: 229px;
  height: 216px;
  padding: 24px 24px 40px 24px;
  @media (max-width: 768px) {
    flex: 1 0 auto;
    margin: 2rem;
  }
`;
