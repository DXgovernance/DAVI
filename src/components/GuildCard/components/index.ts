export { GuildCardHeader } from './GuildCardHeader';
export { GuildCardContent } from './GuildCardContent';
