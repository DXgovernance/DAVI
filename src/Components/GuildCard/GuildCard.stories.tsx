import { ComponentStory, ComponentMeta } from '@storybook/react';
import GuildCard from 'Components/GuildCard/GuildCard';
import { validAddress, nullAddress } from './fixture';
import GuildCardHeader from './GuildCardHeader';
import GuildCardContent from './GuildCardContent';

export default {
  title: 'GuildCard',
  component: GuildCard,
} as ComponentMeta<typeof GuildCard>;

const Template: ComponentStory<typeof GuildCard> = args => (
  <GuildCard {...args}>
    <GuildCardHeader {...args}>Guild Card Header</GuildCardHeader>
    <GuildCardContent>Guild Card Content</GuildCardContent>
  </GuildCard>
);

export const ValidAddress = Template.bind({});
ValidAddress.args = validAddress;

export const NullAddress = Template.bind({});
NullAddress.args = nullAddress;
