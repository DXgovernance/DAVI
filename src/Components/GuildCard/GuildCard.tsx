import UnstyledLink from 'Components/Primitives/Links/UnstyledLink';
import { Card } from 'old-components/Guilds/common/Card/index';
// import { GuildCardProps } from './types';
import GuildCardHeader from './GuildCardHeader';
import GuildCardContent from './GuildCardContent';
import { cardWrapperStyles } from './GuildCard.styled';

interface GuildCardProps {
  guildAddress: string;
  numberOfMembers: any;
  t: any;
  numberOfActiveProposals: any;
  ensName: string;
  data: any;
}

const GuildCard: React.FC<GuildCardProps> = ({
  guildAddress,
  numberOfMembers,
  t,
  numberOfActiveProposals,
  ensName,
  data,
}) => {
  return (
    <UnstyledLink
      data-testid="guildCard"
      to={location => `${location.pathname}/${guildAddress}`}
    >
      <Card customStyles={cardWrapperStyles}>
        <GuildCardHeader
          guildAddress={guildAddress}
          numberOfMembers={numberOfMembers}
          t={t}
          numberOfActiveProposals={numberOfActiveProposals}
        />
        <GuildCardContent
          guildAddress={guildAddress}
          ensName={ensName}
          data={data}
        />
      </Card>
    </UnstyledLink>
  );
};

export default GuildCard;
