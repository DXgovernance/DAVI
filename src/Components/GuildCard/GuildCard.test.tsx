import GuildCard from './GuildCard';
import { render } from '../../utils/tests';
import '@testing-library/jest-dom';
import { validAddress, nullAddress } from './fixture';

jest.mock('hooks/Guilds/ether-swr/ens/useENSNameFromAddress', () => ({
  __esModule: true,
  default: () => 'test.eth',
}));

jest.mock('hooks/Guilds/ether-swr/guild/useGuildConfig', () => ({
  useGuildConfig: function () {
    return {
      data: {
        name: 'REPGuild',
      },
    };
  },
}));

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: key => key,
    i18n: { changeLanguage: jest.fn() },
  }),
}));

jest.mock('hooks/Guilds/ether-swr/guild/useGuildActiveProposals', () => ({
  __esModule: true,
  default: function () {
    return {
      data: {
        _hex: '0x01',
        _isBigNumber: true,
      },
    };
  },
}));

jest.mock('hooks/Guilds/ether-swr/guild/useGuildMemberTotal', () => ({
  __esModule: true,
  default: function () {
    return {
      data: {
        _hex: '0x01',
        _isBigNumber: true,
      },
    };
  },
}));

describe('GuildCard', () => {
  it('Should render properly with a guild address', async () => {
    // const { container } = render(<GuildCard {...validAddress} />);
    // expect(container).toMatchSnapshot();
  });

  it('Should render loading when the address is null', async () => {
    // const { container } = render(<GuildCard {...nullAddress} />);
    // expect(container).toMatchSnapshot();
  });
});
