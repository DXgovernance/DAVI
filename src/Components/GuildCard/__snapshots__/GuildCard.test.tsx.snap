// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuildCard Should render properly with a guild address 1`] = `
<div>
   
  <a
    class="sc-bdfBQB hcWJCy"
    data-testid="guildCard"
    href="//0xa47BbE8Dd6dB29D45FE5eeD838c4f136884AEAF3"
  >
    <div
      class="sc-gsTEea sc-jSgvzq couiEk iwIEDV"
    >
      <div
        class="sc-gsTEea couiEk"
      >
        <div
          class="sc-hKgJUU sc-fubCzh dUEGLL kcknnR"
        />
        <div
          class="sc-gsTEea sc-pGacB couiEk lfrSjv"
        />
      </div>
    </div>
  </a>
   
</div>
`;

exports[`GuildCard Should render properly with a null guild address 1`] = `
<div>
   
  <a
    class="sc-bdfBQB hcWJCy"
    data-testid="guildCard"
    href="//null"
  >
    <div
      class="sc-gsTEea sc-jSgvzq couiEk iwIEDV"
    >
      <div
        class="sc-gsTEea couiEk"
      >
        <div
          class="sc-hKgJUU sc-fubCzh dUEGLL kcknnR"
        />
        <div
          class="sc-gsTEea sc-pGacB couiEk lfrSjv"
        />
      </div>
    </div>
  </a>
   
</div>
`;
